QLogger.cpp

```diff
@@ -1,266 +1,252 @@
+                                    const QString &fileFolderDestination, LogMode mode,
+                                    LogFileDisplay fileSuffixIfFull, LogMessageDisplays messageOptions, bool notify)
 {
-   switch (level)
-   {
-      case LogLevel::Trace:
-         return "Trace";
-      case LogLevel::Debug:
-         return "Debug";
-      case LogLevel::Info:
-         return "Info";
-      case LogLevel::Warning:
-         return "Warning";
-      case LogLevel::Error:
-         return "Error";
-      case LogLevel::Fatal:
-         return "Fatal";
-   }
+   QMutexLocker lock(&mMutex);
 
-   return QString();
-}
-
-bool QLoggerManager::addDestination(const QString &fileDest, const QString &module, LogLevel level)
-{
-   if (!moduleDest.contains(module))
+   if (!mModuleDest.contains(module))
    {
-      const auto log = new QLoggerWriter(fileDest, level);
+       const auto log = new QLoggerWriter((fileDest.isEmpty() ? mDefaultFileDestination : fileDest),
+                                          (level == LogLevel::Warning ? mDefaultLevel : level),
+                                          (fileFolderDestination.isEmpty() ? mDefaultFileDestinationFolder : QDir::fromNativeSeparators(fileFolderDestination)),
+                                          (mode == LogMode::OnlyFile ? mDefaultMode : mode),
+                                          (fileSuffixIfFull == LogFileDisplay::DateTime ? mDefaultFileSuffixIfFull : fileSuffixIfFull),
+                                          (messageOptions.testFlag(LogMessageDisplay::Default) ? mDefaultMessageOptions : messageOptions));
+      log->setMaxFileSize(mDefaultMaxFileSize);
       log->stop(mIsStop);
 
-      moduleDest.insert(module, log);
+      if (notify) {
+        const auto threadId = QString("%1").arg((quintptr)QThread::currentThread(), QT_POINTER_SIZE * 2, 16, QChar('0'));
+        log->enqueue(QDateTime::currentDateTime(), threadId, module, LogLevel::Info, "", "", -1, "Adding destination!");
+      }
+      mModuleDest.insert(module, log);
+
+      if (mode != LogMode::Disabled)
+        log->start();
 
       return true;
    }
 
    return false;
 }
 
-bool QLoggerManager::addDestination(const QString &fileDest, const QStringList &modules, LogLevel level)
+bool QLoggerManager::addDestination(const QString &fileDest, const QStringList &modules, LogLevel level,
+                                    const QString &fileFolderDestination, LogMode mode,
+                                    LogFileDisplay fileSuffixIfFull, LogMessageDisplays messageOptions, bool notify)
 {
+   QMutexLocker lock(&mMutex);
    bool allAdded = false;
 
    for (const auto &module : modules)
    {
-      if (!moduleDest.contains(module))
+      if (!mModuleDest.contains(module))
       {
-         const auto log = new QLoggerWriter(fileDest, level);
+          const auto log = new QLoggerWriter((fileDest.isEmpty() ? mDefaultFileDestination : fileDest),
```

Comment:
We could use the above method in a call in here (removing the lock) and we would avoid all this duplicated code, right?

Reply:
Yes why not. What do you think of creating a new private method called `initializeWriter()` for example ?

Reply:
That is actually a better approach! 👍